### PR TITLE
Pod spec v3, OCI image, team charm reivew

### PR DESCRIPTION
**Pull request by justinmclark**
Like the title says, there are two main changes:

1. Convert pod spec from V1 (default setting) to V3.
2. Use resources rather than config for fetching upstream Grafana image.

**This PR will now include review items from the team Grafana review meeting on November 3, 2020.**

### CODE REVIEW

**Review by balbirthomas — APPROVED** [review]

File: src/charm.py [hunk 1]
```diff
@@ -128,17 +124,12 @@ def has_db(self) -> bool:
         """Only consider a DB connection if we have config info."""
         return len(self.datastore.database) > 0
 
-    def on_config_changed(self, event):
+    def on_config_changed(self, _):
```
[the remark this answers is not in the document]
Comment (justinmclark) [reply src/charm.py:127]: This is exactly why I changed `event` to `_`. I did the same thing on several other functions.

**Review by balbirthomas — APPROVED** [review]
Awesome

File: src/charm.py [hunk 2]
```diff
@@ -178,24 +158,21 @@ def on_grafana_source_changed(self, event):
...
+                        "Using safe default: {}.".format(default_source_name))
+            datasource_fields['source-name'] = default_source_name
+
+        self.datastore.source_names.add(datasource_fields['source-name'])
 
         # set the first grafana-source as the default (needed for pod config)
-        # if `self.datastore.sources` is currently emtpy, this is the first
-        # TODO: confirm that this is what we want
+        # if `self.datastore.sources` is currently empty, this is the first
         if not dict(self.datastore.sources):
```
Comment (Abuelodelanada) [inline src/charm.py:176]: What about avoiding the else:

```python
datasource_fields['isDefault'] = 'false'

if not dict(self.datastore.sources):
    datasource_fields['isDefault'] = 'true'

```

Comment (justinmclark) [reply src/charm.py:176]: Good call. I'll make the change.

**Review by balbirthomas — APPROVED** [review]

**Review by Abuelodelanada — APPROVED** [review]
Looks good to me.

**Review by dstathis — APPROVED** [review]

**Pull request merged**